src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/net461/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets

```diff
@@ -0,0 +1,47 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildThisFileDirectory)..\common.targets" />
+  <Target
+      Name="MvcRazorPrecompile"
+      DependsOnTargets="_ResolveInputArguments"
+      Inputs="$(MSBuildThisFileFullPath);@(MvcRazorFilesToCompile);@(IntermediateAssembly);@(DocFileItem);@(_DebugSymbolsIntermediatePath);@(ReferencePath);$(MSBuildAllProjects);"
```

Comment:
Sorry, let me clarify. 

The reason for using MSBuildAllProjects is to invalidate target caches when project files change. If there is anything down stream from this target (i.e. publishing), you should add this file to the MSBuildAllProjects list. This is a best practice when writing incremental build targets that chain into other targets you don't control.